### PR TITLE
feat: extra attributes for windows

### PR DESCRIPTION
**Pull request by ileodo**
- Adding Window as a binary_sensor (similar to door)
- Adding extra attributes to the window

### CODE REVIEW

File: custom_components/tesla_custom/binary_sensor.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ async def async_setup_entry(hass: HomeAssistant, config_entry, async_add_entitie
         entities.append(TeslaCarChargerConnection(hass, car, coordinator))
         entities.append(TeslaCarCharging(hass, car, coordinator))
         entities.append(TeslaCarDoors(hass, car, coordinator))
+        entities.append(TeslaCarWindows(hass, car, coordinator))
```
Comment (ileodo) [inline custom_components/tesla_custom/binary_sensor.py:36]:
```suggestion
        entities.append(TeslaCarWindows(hass, car, coordinator))
        
```

File: custom_components/tesla_custom/binary_sensor.py [hunk 2]
```diff
@@ -33,6 +33,8 @@ async def async_setup_entry(hass: HomeAssistant, config_entry, async_add_entitie
...
+
```
Comment (ileodo) [inline custom_components/tesla_custom/binary_sensor.py:37]:
```suggestion
        entities.append(TeslaCarWindows(hass, car, coordinator))
```

File: custom_components/tesla_custom/binary_sensor.py [hunk 3]
```diff
@@ -33,6 +33,8 @@ async def async_setup_entry(hass: HomeAssistant, config_entry, async_add_entitie
...
         entities.append(TeslaCarScheduledCharging(hass, car, coordinator))
```
Comment (ileodo) [inline custom_components/tesla_custom/binary_sensor.py:38]:
```suggestion
        entities.append(TeslaCarWindows(hass, car, coordinator))
        entities.append(TeslaCarScheduledCharging(hass, car, coordinator))
```

**Comment by alandtse** [comment]
Thanks for the PR. Please resolve the testing issues.

File: tests/mock_data/car.py [hunk 4]
```diff
@@ -197,7 +197,7 @@
         "dashcam_state": "<invalid>",
         "df": 1,
         "dr": 0,
-        "fd_window": 0,
+        "fd_window": 1,
```
Comment (ileodo) [inline tests/mock_data/car.py:200]:
```suggestion
        "fd_window": 0,
```

File: tests/test_binary_sensor.py [hunk 5]
```diff
@@ -163,6 +163,21 @@ async def test_car_doors(hass: HomeAssistant) -> None:
...
+
+    state = hass.states.get("binary_sensor.my_model_s_windows")
+    assert state.state == STATE_ON
+
+    assert state.attributes.get(ATTR_DEVICE_CLASS) == BinarySensorDeviceClass.WINDOW
+
+    assert state.attributes.get("Driver Front") == "Open"
+    assert state.attributes.get("Driver Rear") == "Closed"
+    assert state.attributes.get("Passenger Front") == "Closed"
+    assert state.attributes.get("Passenger Rear") == "Closed"
```
Comment (ileodo) [inline tests/test_binary_sensor.py:178]:
```suggestion
    state = hass.states.get("binary_sensor.my_model_s_windows")
    assert state.state == STATE_OFF

    assert state.attributes.get(ATTR_DEVICE_CLASS) == BinarySensorDeviceClass.WINDOW

    assert state.attributes.get("Driver Front") == "Closed"
    assert state.attributes.get("Driver Rear") == "Closed"
    assert state.attributes.get("Passenger Front") == "Closed"
    assert state.attributes.get("Passenger Rear") == "Closed"
```

**Comment by ileodo** [comment]
@alandtse the test step in CI is not running.

**Pull request merged**